src/engine/engine.cpp

```diff
@@ -60,9 +52,26 @@ namespace osrm
 namespace engine
 {
 
+// Pointer-to-Impl. hiding target, disallow all other mis-uses
+
+// Not Moveable
+Engine::Engine(Engine &&) noexcept = delete;
+Engine &Engine::operator=(Engine &&) noexcept = delete;
+
+// Not Copyable
+Engine::Engine(const Engine &) = delete;
+Engine &Engine::operator=(const Engine &) = delete;
```

Comment:
These needs to be expanded in the impl. file so the compiler knows about the type's sizes.

See Item 22 in Effective Modern C++ for detailed explanation.